pig/src/test/java/org/jboss/pnc/bacon/pig/impl/addons/scanservice/ValidationTest.java

```diff
@@ -0,0 +1,167 @@
+        try {
+            scanService.validateMapKeys(benchmarkScanServiceConfigMap);
+            Assertions.assertTrue(true);
+        } catch (Exception e) {
+            Assertions.fail();
+        }
+    }
+
+    @Test
+    void negativeKeywordsTest() {
+        HashMap<String, Object> shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("productID");
+        shallowCopy.put("product-id", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("eventId");
+        shallowCopy.put("event-id", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("isManagedService");
+        shallowCopy.put("is-managed-service", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("cpaasVersion");
+        shallowCopy.put("CpaasVersion", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("jobUrl");
+        shallowCopy.put("xyz", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("serviceUrl");
+        shallowCopy.put("service-url", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("serviceSecretKey");
+        shallowCopy.put("service-key", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("serviceSecretValue");
+        shallowCopy.put("service-val", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("brewBuilds");
+        shallowCopy.put("brew_builds", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("scmUrl");
+        shallowCopy.put("SCMURL", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("scmRevision");
+        shallowCopy.put("scm-revision", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("extraScmUrls");
+        shallowCopy.put("xtraScmUrls", null);
+        negativeTest(shallowCopy);
+
+        shallowCopy = new LinkedHashMap<>(benchmarkScanServiceConfigMap);
+        shallowCopy.remove("extraBuilds");
+        shallowCopy.put("extraBuild_", null);
+        negativeTest(shallowCopy);
+    }
+
+    void negativeTest(HashMap<String, Object> testMap) {
```

Comment:
IMO it should be treated as standalone since it consists of multiple 'micro-tests' which differ by a keyword to check only. I agree that the helper method negativeTest() called by all 'micro-tests' should be private rather than package private. I will piggy-back this fix on the next commit.

Reply:
It looks like you copied the same code, but with a different parameter. You could avoid code duplication by moving common code into the method and then using something like

```
@ParameterizedTest
@ValueSource(strings = {"product-id", "event-id", ...})
```

Reply:
Done